app/models/concerns/sequenced.rb

```diff
@@ -32,20 +40,50 @@ def generate_sequential_id
         end
       end
 
-      # NOTE: If the application was unable to aquire the lock, the block returns false
-      raise(SequenceError, 'Unable to aquire lock on the database') unless result
+      # NOTE: If the application was unable to acquire the lock, the block returns false
+      raise(SequenceError, 'Unable to acquire lock on the database') unless result
+
+      result
+    end
+
+    def generate_organization_sequential_id
+      result = Invoice.with_advisory_lock(
+        'invoice_lock',
```

Comment:
I guess we will end with the issue we have faced today. This will lock the entire Invoice table for all invoice generation. Any chance we could scope the lock key to the organization?